kernel/src/main/java/org/kframework/compile/checks/CheckRHSVariables.java

```diff
@@ -29,33 +30,37 @@
  */
 public class CheckRHSVariables {
     private final Set<KEMException> errors;
+    private final boolean errorExistential;
 
-    public CheckRHSVariables(Set<KEMException> errors) {
+    public CheckRHSVariables(Set<KEMException> errors, boolean errorExistential) {
         this.errors = errors;
+        this.errorExistential = errorExistential;
     }
     private void check(Rule rule) {
         resetVars();
         Set<String> unboundVariableNames = getUnboundVarNames(rule);
-        gatherVars(true, rule.body());
-        gatherVars(false, rule.requires());
-        gatherVars(false, rule.ensures());
+        boolean isMacro = ExpandMacros.isMacro(rule);
+        boolean errorExistential = !isMacro && this.errorExistential && !(rule.att().contains(Att.LABEL()) && rule.att().get(Att.LABEL()).equals("STDIN-STREAM.stdinUnblock"));
```

Comment:
Why are `macro` allowed to have existential variables on the RHS?

Please add test-cases exercising each of the clauses in this big conjunct.